WordPress/Classes/Services/MediaService.m

```diff
@@ -376,6 +375,75 @@ - (NSError *)translateMediaUploadError:(NSError *)error {
     return newError;
 }
 
+- (void)deleteMedia:(nonnull Media *)media
+            success:(nullable void (^)())success
+            failure:(nullable void (^)(NSError * _Nonnull error))failure
+{
+    id<MediaServiceRemote> remote = [self remoteForBlog:media.blog];
+    RemoteMedia *remoteMedia = [self remoteMediaFromMedia:media];
+    NSManagedObjectID *mediaObjectID = media.objectID;
+
+    void (^successBlock)() = ^() {
+        [self.managedObjectContext performBlock:^{
+            Media *mediaInContext = (Media *)[self.managedObjectContext existingObjectWithID:mediaObjectID error:nil];
+            [self.managedObjectContext deleteObject:mediaInContext];
+            [[ContextManager sharedInstance] saveContext:self.managedObjectContext
+                                     withCompletionBlock:^{
+                                         if (success) {
+                                             success();
+                                         }
+                                     }];
+        }];
+    };
+    
+    [remote deleteMedia:remoteMedia
+                success:successBlock
+                failure:failure];
+}
+
+- (void)deleteMultipleMedia:(nonnull NSArray<Media *> *)mediaObjects
+                   progress:(nullable void (^)(NSProgress *_Nonnull progress))progress
+                    success:(nullable void (^)())success
+                    failure:(nullable void (^)())failure
+{
+    if (mediaObjects.count == 0) {
+        if (success) {
+            success();
+        }
+        return;
+    }
+
+    NSProgress *currentProgress = [NSProgress progressWithTotalUnitCount:mediaObjects.count];
+
+    dispatch_group_t group = dispatch_group_create();
+
+    [mediaObjects enumerateObjectsUsingBlock:^(Media *media, NSUInteger idx, BOOL *stop) {
+        dispatch_group_enter(group);
+        [self deleteMedia:media success:^{
```

Comment:
This looks like case to reconsider implementing back support for batch calls in the REST API . :)